tests/periph_uart/main.c

```diff
@@ -259,12 +320,41 @@ static int cmd_send(int argc, char **argv)
     return 0;
 }
 
+static int cmd_test(int argc, char **argv)
+{
+    int dev;
+
+    if (argc < 2) {
+        printf("usage: %s <dev>\n", argv[0]);
+        return 1;
+    }
+    /* parse parameters */
+    dev = parse_dev(argv[1]);
+    if (dev < 0) {
+        return 1;
+    }
+
+    puts("[START]");
+
+    /* run self test with different baud rates */
+    for (unsigned i = 1; i <= 12; ++i) {
+        if (_self_test(dev, 9600 * i)) {
+            puts("[FAILURE]");
+            return -1;
+        }
+    }
+
+    puts("[SUCCESS]");
+    return 0;
+}
+
 static const shell_command_t shell_commands[] = {
     { "init", "Initialize a UART device with a given baudrate", cmd_init },
 #ifdef MODULE_PERIPH_UART_MODECFG
     { "mode", "Setup data bits, stop bits and parity for a given UART device", cmd_mode },
 #endif
     { "send", "Send a string through given UART device", cmd_send },
+    { "test", "Run an automated test on a UART with RX and TX connected", cmd_test },
```

Comment:
I really like this `test` command! It really help during setting up the board config for a second UART.